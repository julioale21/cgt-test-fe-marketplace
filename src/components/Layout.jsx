import { Outlet } from 'react-router-dom';
import { Header } from '../components/shared/Header';
import { Box } from '@mui/material';
import { useCart } from '../hooks/useCart';

const Layout = () => {
  const { items } = useCart();

  const cartItemsCount = items.reduce((total, item) => total + item.quantity, 0);

  return (
    <Box sx={{ display: 'flex', flexDirection: 'column', minHeight: '100vh' }}>
      <Header cartItemsCount={cartItemsCount} />

      <Box component="main" sx={{ flexGrow: 1, p: 3 }}>
        <Outlet />
      </Box>

      <Box component="footer" sx={{ p: 2, textAlign: 'center', mt: 'auto' }}>
        <p>Â© 2025</p>
      </Box>
    </Box>
  );
};

export { Layout };
