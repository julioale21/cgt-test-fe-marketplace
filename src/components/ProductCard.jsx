import React from 'react';

import { Link } from 'react-router-dom';

import { Button, Card, CardActions, CardContent, CardMedia, Grid, Typography } from '@mui/material';
import ShoppingCartIcon from '@mui/icons-material/ShoppingCart';

const ProductCard = ({ product }) => {
  return (
    <Grid item xs={12} sm={6} md={4} lg={3}>
      <Card sx={{ height: '100%', display: 'flex', flexDirection: 'column' }}>
        <CardMedia
          component="img"
          height="200"
          image={product.image}
          alt={product.name}
          sx={{
            objectFit: 'cover',
            height: '200px'
          }}
        />
        <CardContent sx={{ flexGrow: 1 }}>
          <Typography gutterBottom variant="h6" component="h2">
            {product.name}
          </Typography>
          <Typography variant="body2" color="text.secondary" sx={{ mb: 2 }}>
            {product.description}
          </Typography>
          <Typography variant="h6" color="primary" sx={{ mt: 2 }}>
            ${product.price.toFixed(2)}
          </Typography>
          <Typography variant="body2" color="text.secondary" sx={{ mt: 1 }}>
            Formats: {product.format.join(', ')}
          </Typography>
        </CardContent>
        <CardActions sx={{ justifyContent: 'space-between' }}>
          <Button component={Link} to={`/products/${product.id}`} size="small" color="primary">
            View Details
          </Button>
          <Button size="small" variant="contained" color="primary" startIcon={<ShoppingCartIcon />}>
            Add to Cart
          </Button>
        </CardActions>
      </Card>
    </Grid>
  );
};

export { ProductCard };
