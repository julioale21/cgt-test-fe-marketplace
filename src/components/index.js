export { CartHeader } from "./cart/CartHeader";
export { CartItem } from "./cart/CartItem";
export { CartItemList } from "./cart/CartItemList";
export { CartSummary } from "./cart/CartSummary";
export { CartSummaryRow } from "./cart/CartSummaryRow";
export { ContainedButton } from "./buttons/ContainedButton";
export { CustomAlert } from "./shared/CustomAlert";
export { EmptyCart } from "./cart/EmptyCart";
export { Header } from "./shared/Header";
export { Layout } from "./Layout";
export { OutlinedButton } from "./buttons/OutlinedButton";
export { PaymentMethodIcon } from "./cart/PaymentMethodIcon";
export { ProductCard } from "./product/ProductCard";
export { ProductDetails } from "./product/ProductDetail";
export { ProductImage } from "./product/ProductImage";
export { ProductList } from "./product/ProductList";
export { ProductSpecs } from "./product/ProductSpecs";
export {ProductHeader} from "./product/ProductHeader";
