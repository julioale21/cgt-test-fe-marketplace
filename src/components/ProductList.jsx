import React from 'react';
import { Grid } from '@mui/material';
import { ProductCard } from './ProductCard';

const ProductList = ({ products }) => {
  return (
    <Grid container spacing={4}>
      {products.map((product) => (
        <ProductCard key={product.id} product={product} />
      ))}
    </Grid>
  );
};

export { ProductList };
