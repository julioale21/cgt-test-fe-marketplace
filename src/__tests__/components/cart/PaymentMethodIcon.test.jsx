import React from 'react';
import { render, screen } from '@testing-library/react';
import { PaymentMethodIcon } from '../../../components';

describe('PaymentMethodIcon', () => {
  const defaultProps = {
    src: 'test-image.png',
    alt: 'Test payment method'
  };

  it('matches snapshot', () => {
    const { container } = render(<PaymentMethodIcon {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it('renders with default size', () => {
    render(<PaymentMethodIcon {...defaultProps} />);
    const img = screen.getByRole('img');

    expect(img).toHaveStyle({
      height: '60px',
      width: '60px'
    });
  });

  it('applies custom size within valid range', () => {
    render(<PaymentMethodIcon {...defaultProps} size={100} />);
    const img = screen.getByRole('img');

    expect(img).toHaveStyle({
      height: '100px',
      width: '100px'
    });
  });

  it('clamps size to minimum value', () => {
    render(<PaymentMethodIcon {...defaultProps} size={10} />);
    const img = screen.getByRole('img');

    expect(img).toHaveStyle({
      height: '20px',
      width: '20px'
    });
  });

  it('clamps size to maximum value', () => {
    render(<PaymentMethodIcon {...defaultProps} size={300} />);
    const img = screen.getByRole('img');

    expect(img).toHaveStyle({
      height: '200px',
      width: '200px'
    });
  });

  it('returns null when src is not provided', () => {
    const consoleSpy = jest.spyOn(console, 'warn').mockImplementation(() => {});
    const { container } = render(<PaymentMethodIcon alt="Test" />);

    expect(container.firstChild).toBeNull();
    expect(consoleSpy).toHaveBeenCalledWith(
      'PaymentMethodIcon: src prop is required for proper rendering'
    );

    consoleSpy.mockRestore();
  });

  it('applies grayscale filter and opacity by default', () => {
    render(<PaymentMethodIcon {...defaultProps} />);
    const img = screen.getByRole('img');

    expect(img).toHaveStyle({
      filter: 'grayscale(100%)',
      opacity: '0.7'
    });
  });

  it('includes hover opacity styles', () => {
    const { container } = render(<PaymentMethodIcon {...defaultProps} />);
    const styles = window.getComputedStyle(container.firstChild);

    expect(styles).toBeDefined();
  });

  it('maintains object-fit contain property', () => {
    render(<PaymentMethodIcon {...defaultProps} />);
    const img = screen.getByRole('img');

    expect(img).toHaveStyle({
      objectFit: 'contain'
    });
  });
});
