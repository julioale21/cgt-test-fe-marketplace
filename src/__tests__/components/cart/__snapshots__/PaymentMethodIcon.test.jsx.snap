// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentMethodIcon matches snapshot 1`] = `
<div>
  <img
    alt="Test payment method"
    class="MuiBox-root css-tv7otk"
    src="test-image.png"
  />
</div>
`;
